code/modules/shuttle/emergency.dm

```diff
@@ -75,6 +102,98 @@
+		increase_hijack_stage()
+		console_hijack_turf = get_turf(src)
+		message_admins("[ADMIN_LOOKUPFLW(user)] has hijacked [src] in [ADMIN_VERBOSEJMP(console_hijack_turf)]. Hijack stage increased to stage [SSshuttle.emergency.hijack_status] out of [HIJACKED].")
+		log_game("[key_name(usr)] has hijacked [src]. Hijack stage increased to stage [SSshuttle.emergency.hijack_status] out of [HIJACKED].")
+		. = TRUE
+		to_chat(user, "<span class='notice'>You reprogram some of [src]'s programming, putting it on timeout for [hijack_stage_cooldown/10] seconds.</span>")
+		visible_message("<span class='warning'>[user.name] appears to be tampering with [src].</span>")
+	hijack_hacking = FALSE
+
+/obj/machinery/computer/emergency_shuttle/proc/announce_hijack_stage()
+	var/msg
+	switch(SSshuttle.emergency.hijack_status)
+		if(NOT_BEGUN)
+			return
+		if(STAGE_1)
+			if(ishuman(usr))
+				var/mob/living/carbon/human/H = usr
+				msg = "AUTHENTICATING - FAIL. AUTHENTICATING - FAIL. AUTHENTICATING - FAI###### Welcome, technician: [random_name(pick(MALE, FEMALE), H.dna.species.name)]. Debug mode: Enabled."
+			else
+				msg = "AUTHENTICATING - FAIL. AUTHENTICATING - FAIL. AUTHENTICATING - FAI###### Welcome, technician: admin_[rand(0,25)]. Debug mode: Enabled."
+		if(STAGE_2)
+			msg = "Warning: Navigational route fails \"IS_AUTHORIZED\". Please try againNN[Gibberish("againagainerroragainagain", 70, 50)]."
+		if(STAGE_3)
+			msg = "CRC mismatch at ~h~ in calculated route buffer. Full reset initiated of FTL_NAVIGATION_SERVICES. Memory decrypted for automatic repair."
+		if(STAGE_4)
+			msg = "~ACS_directive module_load(cyberdyne.exploit.nanotrasen.shuttlenav)... NT key mismatch. Confirm load? Y...###Reboot complete. CALL transponder.disable(1) ON /transmitter IN world; System link initiated with connected engines..."
+		if(HIJACKED)
+			msg = "SYSTEM OVERRIDE - Resetting course to \[[Gibberish("###########", 100, 90)]\] \
+			([Gibberish("#######", 100, 90)]/[Gibberish("#######", 100, 90)]/[Gibberish("#######", 100, 90)]) \
+			{AUTH - ROOT (uid: 0)}. <br>\
+			[SSshuttle.emergency.mode == SHUTTLE_ESCAPE ? "Diverting from existing route - Bluespace exit in <br>\
+			[hijack_completion_flight_time_set >= INFINITY ? "[Gibberish("\[ERROR\]", 50, 50)]" : hijack_completion_flight_time_set/10] seconds." : ""]"
+	announce_here("SYSTEM ERROR", Gibberish(msg, 70, rand(3,6)))
+
+
+/obj/machinery/computer/emergency_shuttle/proc/announce_here(a_header = "Emergency Shuttle", a_text = "")
+	var/area/A = get_area(src)
+	var/msg_text = "<b><font size=4 color=red>[a_header]</font><br> <font size=3><span class='robot'>[a_text]</font size></font></b></span>"
+	var/list/receivers = list()
+	for(var/mob/M in GLOB.mob_list)
+		if(!M.ckey)
+			continue
+		var/turf/T = get_turf(M)
+		if(T && T.loc && T.loc == A)
+			receivers |= M
+	for(var/mob/R in receivers)
+		to_chat(R, msg_text)
+		SEND_SOUND(R, sound('sound/misc/notice1.ogg'))
```

Comment:
I am not fond of this. you could just do `for(var/mob/reciever in A)`. the only issue with this, is mobs in lockers cant hear it. alternatively you could just do for(var/mob/R in range(howbig the shuttle is)) or use play sound.

Reply:
I'll just do range, so mechs and lockered people can hear it.

Reply:
range done